wasmsdk/blobber.go

```diff
@@ -333,6 +336,154 @@ func download(allocationID, remotePath, authTicket, lookupHash string, downloadT
 
 }
 
+type BulkUploadOption struct {
+	AllocationID string `json:"allocationID,omitempty"`
+	RemotePath   string `json:"remotePath,omitempty"`
+
+	ThumbnailBytes []byte `json:"thumbnailBytes,omitempty"`
+	Encrypt        bool   `json:"encrypt,omitempty"`
+	IsUpdate       bool   `json:"isUpdate,omitempty"`
+	IsRepair       bool   `json:"isRepair,omitempty"`
+
+	NumBlocks         int    `json:"numBlocks,omitempty"`
+	FileSize          int64  `json:"fileSize,omitempty"`
+	ReadChunkFuncName string `json:"readChunkFuncName,omitempty"`
+	CallbackFuncName  string `json:"callbackFuncName,omitempty"`
+}
+
+type BulkUploadResult struct {
+	RemotePath string `json:"remotePath,omitempty"`
+	Success    bool   `json:"success,omitempty"`
+	Error      string `json:"error,omitempty"`
+}
+
+func bulkUpload(jsonBulkUploadOptions string) ([]BulkUploadResult, error) {
+	var options []BulkUploadOption
+	err := json.Unmarshal([]byte(jsonBulkUploadOptions), &options)
+	if err != nil {
+		return nil, err
+	}
+
+	for _, option := range options {
+		go func(o BulkUploadOption) {
+			ok, err := uploadWithJsFuncs(o.AllocationID, o.RemotePath,
+				o.ReadChunkFuncName,
+				o.FileSize,
+				o.ThumbnailBytes,
+				o.Encrypt,
+				o.IsUpdate,
+				o.IsRepair,
+				o.NumBlocks,
+				o.CallbackFuncName)
+			result := BulkUploadResult{
+				RemotePath: o.RemotePath,
+				Success:    ok,
+			}
+			if err != nil {
+				result.Error = err.Error()
+				result.Success = false
+			}
+		}(option)
+
+	}
+
+	n := len(options)
+	results := make([]BulkUploadResult, 0, n)
+	wait := make(chan BulkUploadResult, 1)
+
+	for i := 0; i < n; i++ {
+		result := <-wait
```

Comment:
There's no writer of the `wait` channel, it will get stuck here. We need to move the push the result to the wait channel above in the goroutine above.

Reply:
fixed